src/test/java/pl/edu/agh/firecell/engine/algorithm/BasicAlgorithmTest.java

```diff
@@ -0,0 +1,82 @@
+package pl.edu.agh.firecell.engine.algorithm;
+
+import org.joml.Vector3i;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import pl.edu.agh.firecell.model.Cell;
+import pl.edu.agh.firecell.model.Material;
+import pl.edu.agh.firecell.model.State;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Stream;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class BasicAlgorithmTest {
+
+    @ParameterizedTest(name = "Compute")
+    @MethodSource("engineComputeTestCases")
+    public void flattenIndexTest(State state, Vector3i cellIndex, Cell resultCell, BasicAlgorithm algorithm) {
+        // given when then
+        assertEquals(algorithm.compute(state, cellIndex), resultCell);
+    }
+
+    public static Stream<Arguments> engineComputeTestCases() {
+
+        double deltaTime = 0.5;
+        Vector3i spaceSize = new Vector3i(3, 3, 3);
+
+        State initWoodenState = getInitState(spaceSize, Material.WOOD);
+        State initAirState = getInitState(spaceSize, Material.AIR);
+
+        Vector3i testCase1Wood = new Vector3i(1, 1, 1);
+        Vector3i testCase2Wood = new Vector3i(2, 0, 1);
+        Vector3i testCase3Wood = new Vector3i(2, 2, 2);
+
+        Vector3i testCase1Air = new Vector3i(1, 1, 1);
+        Vector3i testCase2Air = new Vector3i(2, 0, 1);
+        Vector3i testCase3Air = new Vector3i(2, 2, 2);
+
+        BasicAlgorithm algorithm = new BasicAlgorithm(deltaTime);
+        double coe = BasicAlgorithm.GAMMA_PRIM_N1;
+
+        return Stream.of(
+                Arguments.of(initWoodenState, testCase1Wood, cell(initWoodenState.getCell(testCase1Wood).temperature() + deltaTime * coe * 400), algorithm),
```

Comment:
Arguments of the test should not be computed (especially the same way our testing method computes them XD).
We can make `deltaTime` and `coe` separate arguments of this test.
But we'll leave it for now.
After merging create a Jira task for fixing it.

Reply:
For now expected result is depend on variables, coe and deltaTime. So I don't know is it good to hard-code them.